src/DynamoCore/DSEngine/AstBuilder.cs

```diff
@@ -83,6 +93,80 @@ public static IEnumerable<NodeModel> TopologicalSort(IEnumerable<NodeModel> node
             return sortedNodes.Where(nodeModels.Contains);
         }
 
+        /// <summary>
+        /// BFS and post-order traverse to ensure all inputs are visited in
+        /// order and before downstream node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="nodeFlags"></param>
+        /// <param name="sortedNodes"></param>
+        private static void BFSTraverse(
+            NodeModel node, 
+            Dictionary<NodeModel, MarkFlag> nodeFlags, 
+            Queue<NodeModel> sortedNodes)
+        {
```

Comment:
no, it is a private method and we could ensure the input `node` is not null. But yes i'll add the checking for public method.